## Sales Data Analysis

We have downloaded a sample sales dataset from <a href='https://www.kaggle.com/datasets/beekiran/sales-data-analysis'>Kaggle.com</a> in order to make a demo analysis with Python. First, we download the file to our computer and upload it to the jupyter notebook.
In addition, we import the required python library to conduct the analysis.

In [1]:
import pandas as pd

We will have an overview of our data, to see what we are dealing with

In [2]:
sales = pd.read_csv('Sales Data.csv')

In [3]:
sales.head()

Unnamed: 0  Order ID               Product  Quantity Ordered  Price Each  \
0           0    295665    Macbook Pro Laptop                 1     1700.00   
1           1    295666    LG Washing Machine                 1      600.00   
2           2    295667  USB-C Charging Cable                 1       11.95   
3           3    295668      27in FHD Monitor                 1      149.99   
4           4    295669  USB-C Charging Cable                 1       11.95   

            Order Date                        Purchase Address  Month  \
0  2019-12-30 00:01:00  136 Church St, New York City, NY 10001     12   
1  2019-12-29 07:03:00     562 2nd St, New York City, NY 10001     12   
2  2019-12-12 18:21:00    277 Main St, New York City, NY 10001     12   
3  2019-12-22 15:13:00     410 6th St, San Francisco, CA 94016     12   
4  2019-12-18 12:38:00           43 Hill St, Atlanta, GA 30301     12   

     Sales            City  Hour  
0  1700.00   New York City     0  
1   600.00   New York City     7  
2    11.95   New York City    18  
3   149.99   San Francisco    15  
4    11.95         Atlanta    12

In [4]:
sales.columns

Index(['Unnamed: 0', 'Order ID', 'Product', 'Quantity Ordered', 'Price Each',
       'Order Date', 'Purchase Address', 'Month', 'Sales', 'City', 'Hour'],
      dtype='object')

In [5]:
print (sales.shape)

(185950, 11)


In the first table, we can see only the first 5 rows.
We can see that the dataset consists of 11 rows and 185950 rows. That's a good start.
We want to make sure that the column "Order Date" is of datetime in order to proceed to our calculations

In [6]:
sales["Order Date"] = pd.to_datetime(sales["Order Date"])

Let's check what type of columns does the dataset consist of

In [7]:
sales.dtypes

Unnamed: 0                   int64
Order ID                     int64
Product                     object
Quantity Ordered             int64
Price Each                 float64
Order Date          datetime64[ns]
Purchase Address            object
Month                        int64
Sales                      float64
City                        object
Hour                         int64
dtype: object

We want to group by City, Product and year, in order to see the top sales per city and year and present them in a form of table.
We will create a new column 'year'

In [8]:
sales['year'] = sales['Order Date'].dt.year

In [9]:
year = sales['Order Date'].dt.year

In [10]:
sales.groupby(['year','City','Product'], as_index=False)['Quantity Ordered'].sum()

year            City                     Product  Quantity Ordered
0    2019         Atlanta                20in Monitor               342
1    2019         Atlanta      27in 4K Gaming Monitor               493
2    2019         Atlanta            27in FHD Monitor               588
3    2019         Atlanta      34in Ultrawide Monitor               482
4    2019         Atlanta       AA Batteries (4-pack)              2193
..    ...             ...                         ...               ...
193  2020   San Francisco  Bose SoundSport Headphones                 2
194  2020   San Francisco                Google Phone                 1
195  2020   San Francisco        USB-C Charging Cable                 1
196  2020   San Francisco                      iPhone                 1
197  2020         Seattle            27in FHD Monitor                 1

[198 rows x 4 columns]

There they are. Now let's see in the dataset, what are all the products that we are dealing with..

In [11]:
print(sales.Product.unique())
print("No of products:",len(pd.unique(sales['Product'])))

['Macbook Pro Laptop' 'LG Washing Machine' 'USB-C Charging Cable'
 '27in FHD Monitor' 'AA Batteries (4-pack)' 'Bose SoundSport Headphones'
 'AAA Batteries (4-pack)' 'ThinkPad Laptop' 'Lightning Charging Cable'
 'Google Phone' 'Wired Headphones' 'Apple Airpods Headphones'
 'Vareebadd Phone' 'iPhone' '20in Monitor' '34in Ultrawide Monitor'
 'Flatscreen TV' '27in 4K Gaming Monitor' 'LG Dryer']
No of products: 19
